### PR TITLE
UI for countdown.

### PR DESCRIPTION
**Pull request by nddmanh**
UI for countdown:
image

### CODE REVIEW

File: src/modules/playRoom/pages/PlayRoom.css [hunk 1]
```diff
@@ -0,0 +1,15 @@
+.countdown {
+    margin-top: 30px;
+    background-color: gray;
```
Comment (lethemanh) [inline src/modules/playRoom/pages/PlayRoom.css]: Use hex code for background color

File: src/modules/playRoom/pages/PlayRoom.css [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    width: 100%;
+    height: 350px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    border-radius: 10px;
+    border: solid 5px blue;
```
Comment (lethemanh) [inline src/modules/playRoom/pages/PlayRoom.css]: Here too

File: src/modules/playRoom/components/CountDown.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const CountDown = () => {
+  return (
+    <div className="countdown">
+      <p className="countdown-time">
+        Thời gian còn lại:
```
Comment (lethemanh) [inline src/modules/playRoom/components/CountDown.js]: For the text ```Thời gian còn lại``` I think should be ```Thời gian còn lại để đặt cược:```, it make text more meaningful

File: src/modules/playRoom/components/CountDown.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const CountDown = () => {
+  return (
+    <div className="countdown">
+      <p className="countdown-time">
+        Thời gian còn lại: 
+        <strong> 59 </strong>
```
Comment (lethemanh) [inline src/modules/playRoom/components/CountDown.js]: Display time unit here

**Review by lethemanh — APPROVED** [review]

File: src/modules/playRoom/pages/PlayRoom.css [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    
+}
+.countdown-time {
+    text-align: center;
```
Comment (huyanh7796) [inline src/modules/playRoom/pages/PlayRoom.css:14]: countdown time's font-size is too small. It should be increased a little bit

File: src/modules/playRoom/pages/PlayRoom.scss [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    background-color: $bg-color-countdown;
+    width: 100%;
+    height: 350px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    border-radius: 10px;
+    border: solid 10px $border-countdown-color;
+    &-time {
+        text-align: center;
```
Comment (huyanh7796) [inline src/modules/playRoom/pages/PlayRoom.scss:194]: countdown time's font-size is too small. It should be increased a little bit

File: src/modules/playRoom/pages/PlayRoom.scss [hunk 7]
```diff
@@ -0,0 +1,17 @@
+$bg-color-countdown: #dfe6e9;
+$border-countdown-color: #77b3d4;
+
+.countdown {
+    margin-top: 30px;
+    background-color: $bg-color-countdown;
+    width: 100%;
+    height: 350px;
```
Comment (ppppbn) [inline src/modules/playRoom/pages/PlayRoom.scss:8]: Too big for a count down container

File: src/modules/playRoom/router/index.js [hunk 8]
```diff
@@ -7,7 +7,7 @@ const authRoutes = [
     name: 'PlayRoom',
     component: PlayRoom,
     meta: {
-      requireAuth: true,
+      requireAuth: false,
```
Comment (ppppbn) [inline src/modules/playRoom/router/index.js]: Remove

File: src/modules/playRoom/pages/PlayRoom.css [hunk 9]
```diff
@@ -0,0 +1,15 @@
+.countdown {
```
Comment (ppppbn) [inline src/modules/playRoom/pages/PlayRoom.css]: Remove the file as you already have PlayRoom.scss

**Comment by nddmanh** [comment]
After resolve:
image

File: src/modules/playRoom/components/CountDown.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState }  from 'react';
+
+const CountDown = () => {
+  const [countDown, setCountDown] = useState(59);
+  useEffect(() => {
+    const time = setInterval(() => {
+      setCountDown(countDown - 1);
+    }, 1000);
+    if (countDown === -1) {
+      setCountDown(59) ;
```
Comment (lethemanh) [inline src/modules/playRoom/components/CountDown.js]: Remove unnecessary whitespace

**Review by huyanh7796 — APPROVED** [review]

File: src/modules/playRoom/components/CountDown.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState }  from 'react';
+
+const CountDown = () => {
+  const [countDown, setCountDown] = useState(59);
+  useEffect(() => {
+    const time = setInterval(() => {
+      setCountDown(countDown - 1);
+    }, 1000);
+    if (countDown === -1) {
```
Comment (lethemanh) [inline src/modules/playRoom/components/CountDown.js]: Please explain for me why you have to setCountDown when ```countDown === -1```

Comment (nddmanh) [reply src/modules/playRoom/components/CountDown.js]: because when the countdown goes to -1, on the display we will see from 0 jump to 59. If I set countdown is 0, on the display we will see from 1 jump to 59

Comment (lethemanh) [reply src/modules/playRoom/components/CountDown.js]: I mean why you have to reset ```countDown``` value to 59, why you don't hide count down and start calculate result

Comment (nddmanh) [reply src/modules/playRoom/components/CountDown.js]: Oh, I didn't think of that. I will resolve it. Thank you.

**Comment by nddmanh** [comment]
When countdown reaches 0, this modal will appear for 5 seconds and disappear:
image

**Review by lethemanh — APPROVED** [review]
